src/logcollector/read_win_event_channel.c

```diff
@@ -551,22 +563,34 @@ void win_start_event_channel(char *evt_log, char future, char *query)
 
     /* Create copy of event log string */
     if ((channel->bookmark_name = strdup(channel->evt_log)) == NULL) {
-        mferror(
+        merror(
             "Could not strdup() event log name to start reading (%s) which returned [(%d)-(%s)]",
             channel->evt_log,
             errno,
             strerror(errno));
         goto cleanup;
     }
 
+    if (query) {
+        /* Create copy of query string */
+        if ((channel->query = strdup(query)) == NULL) {
```

Comment:
`channel->query` is not freed on clean-up. In addition, it is not needed since the query is correctly filtered some lines below and sent to `EvtSubscribe()` as `wquery`